stream_alert/classifier/clients/sqs.py

```diff
@@ -0,0 +1,319 @@
+            message = cls._extract_message_by_id(messages, success_id)
+            if not message:
+                continue
+            messages.remove(message)
+
+    @classmethod
+    def _extract_message_by_id(cls, batch, identifier):
+        """Extract a message from the batch given the # IDEA:
+
+        Args:
+            batch (list): Batch of messages from which to extract the message
+            identifier (str): Id for the message to be extracted
+        """
+        for message in batch:
+            if message['Id'] == identifier:
+                return message
+
+        # Log an error if a message with this ID was not found in the batch
+        LOGGER.error('SQS message with ID \'%s\' not found in batch', identifier)
+
+    def _check_failures(self, response, batch=None):
+        """Inspect the response to see if the failure was our fault (the Sender)
+
+        Args:
+            response (dict): Response object from the boto3.resource.send_messages call
+                that contains metadata on the success status of the call
+            batch (list): List of dicts with JSON dumped records as MessageBody that are being
+                sent to SQS. Format is:
+                [{'Id': '...', 'MessageBody': '...'}, {'Id': '...', 'MessageBody': '...'}]
+        Raises:
+            SQSClientError: Indication that there is something wrong with the sender configuration
+        """
+        if not response.get('Failed'):
+            return 0  # nothing to do here
+
+        LOGGER.error('The following records failed to put to the SQS Queue: %s',
+                     self.queue.url)
+
+        for failure in response['Failed']:
+            # Pull out the record that matches this ID
+            record = self._extract_message_by_id(batch, failure['Id']) if batch else None
+            LOGGER.error(self._format_failure_message(failure, record=record))
+
+        failed = len(response.get('Failed', []))
+        self._log_failed(failed)
+
+        # Raise an exception if this is the fault of the sender (us)
+        if any(result['SenderFault'] for result in response['Failed']):
+            raise SQSClientError('Failed to send records to SQS:\n{}'.format(response))
+
+        return failed
+
+    def _send_messages(self, batched_messages):
+        """Send new formatted messages to CSIRT SQS
+        Args:
+            batched_messages (list): A list of messages that are already serialized to json
+                to be sent to the Rules Engine function
+        Returns:
+            bool: True if the request was successful, False otherwise
+        """
+        @backoff.on_predicate(backoff.fibo,
+                              lambda resp: len(resp.get('Failed', [])) > 0,
+                              max_tries=self.MAX_BACKOFF_ATTEMPTS,
+                              max_value=self.MAX_BACKOFF_FIBO_VALUE,
+                              jitter=backoff.full_jitter,
```

Comment:
This is [default jitter algorithm](https://github.com/litl/backoff/blob/master/backoff/_decorator.py#L15). We can remove this line.

Reply:
good catch!!